src/gc.c

```diff
@@ -3040,6 +3248,146 @@ JL_DLLEXPORT jl_value_t *jl_gc_alloc_3w(void)
     return jl_gc_alloc(ptls, sizeof(void*) * 3, NULL);
 }
 
+JL_DLLEXPORT int jl_gc_enable_conservative_gc_support(void)
+{
+    static_assert(jl_buff_tag % GC_PAGE_SZ == 0,
+        "jl_buff_tag must be a multiple of GC_PAGE_SZ");
+    if (jl_is_initialized()) {
+        int result = jl_atomic_fetch_or(&support_conservative_marking, 1);
```

Comment:
I believe exchange is better but this isn't a performance critical point so either is fine...